### PR TITLE
Documentation on how to use FQDM with NServiceBus

### PR DESCRIPTION
**Pull request by janovesk**

### CODE REVIEW

File: nservicebus/msmq/full-qualified-domain-name.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+title: MSMQ connection strings
+summary: Detailed connection string information for MSMQ.
+tags:
+- Connection strings
+- Transports
+---
+
+
+
+NServiceBus with MSMQ use the local machine name, taken from `Environment.MachineName`, as  as the originator and reply-to address in messages by default.
```
Comment (SimonCropp) [inline nservicebus/msmq/full-qualified-domain-name.md]: "as  as"

File: nservicebus/msmq/full-qualified-domain-name.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Some deployment enviroments requires use of [Full Qualified Domain Names(FQDM)](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) to route messages correctly. The most common scenario is routing between machines in different domains. 
+
+To override how NServiceBus resolves the machine name, provide a factory method to  `RuntimeEnvironment.MachineNameAction` when an endpoint is configured.
+
+Check [how to find fqdn of local machine](http://stackoverflow.com/questions/804700/how-to-find-fqdn-of-local-machine-in-c-net) for a good starting point on how to get the FQDM of the local machine.
+
+snippet:MsmqMachineNameFQDN
+
+Remember that as soon as you start using FQDM you probably want to do the same for message mappings and any direct queue destinations you provide to NServiceBus.
```
Comment (SimonCropp) [inline nservicebus/msmq/full-qualified-domain-name.md]: perhaps

> When first adopting FQDM the same will need to be done for message mappings and any direct queue destinations provided to NServiceBus.

File: nservicebus/msmq/full-qualified-domain-name.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+tags:
+- Connection strings
+- Transports
+---
+
+
+
+NServiceBus with MSMQ use the local machine name, taken from `Environment.MachineName`, as  as the originator and reply-to address in messages by default.
+
+Some deployment enviroments requires use of [Full Qualified Domain Names(FQDM)](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) to route messages correctly. The most common scenario is routing between machines in different domains.
```
Comment (SimonCropp) [inline nservicebus/msmq/full-qualified-domain-name.md]: Perhaps

> Some deployment enviroments require the use of

File: nservicebus/msmq/full-qualified-domain-name.md (hunk 1 above)
Comment (SimonCropp) [inline nservicebus/msmq/full-qualified-domain-name.md]: perhaps

> uses the local

File: Snippets/Snippets_4/Transports/Msmq/FQDM.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+{
+    using NServiceBus.Support;
+
+    public class FQDMTest
+    {
+        void FQDM()
+        {
+            #regionMsmqMachineNameFQDN
+
+            RuntimeEnvironment.MachineNameAction = () => System.Net.Dns.GetHostEntry(Environment.MachineName).HostName;
```
Comment (SimonCropp) [inline Snippets/Snippets_4/Transports/Msmq/FQDM.cs]: any 5 and 6 versions?

**Comment by janovesk** [comment]
Please check and merge @SimonCropp. `[4,)` is the right way to specify version 4+, right?

**Pull request merged**